src/basic/unit-def.h

```diff
@@ -5,14 +5,17 @@
 
 #include "macro.h"
 
+/* The enum order is used to order unit jobs in the job queue
+ * when other criteria (cpu weight, nice level) are identical.
+ * In this case service units have the hightest priority. */
 typedef enum UnitType {
         UNIT_SERVICE = 0,
+        UNIT_MOUNT,
+        UNIT_SWAP,
         UNIT_SOCKET,
         UNIT_TARGET,
         UNIT_DEVICE,
-        UNIT_MOUNT,
         UNIT_AUTOMOUNT,
-        UNIT_SWAP,
```

Comment:
Now I'm confused. In the original patch, non-exec units like automount or device or target were ordered earlier. But now, this puts service units first.

Reply:
I think putting exec jobs first makes a lot of sense, since it increases parallelism, but dunno, why would you prefer it the other way round?

Reply:
Preferring targets is only useful if that comparison is more important than the cpu weight /  nice level. That is no longer the case.

Also, @poettering requested that services should be preferred to improve parallelism. I expanded that to all units that fork new processes.